zerver/tests/test_import_export.py

```diff
@@ -443,6 +449,116 @@ def find_by_id(table: str, db_id: int) -> Dict[str, Any]:
+        self.assertIn('default-bot@zulip.com', exported_user_emails)
+        self.assertIn('emailgateway@zulip.com', exported_user_emails)
+
+        exported_streams = get_set('zerver_stream', 'name')
+        self.assertEqual(
+            exported_streams,
+            set([u'Denmark', u'Rome', u'Scotland', u'Venice', u'Verona',
+                 u'Private A', u'Private B', u'Private C'])
+        )
+
+        data = full_data['message']
+        exported_usermessages = UserMessage.objects.filter(user_profile__in=[self.example_user("iago"),
+                                                                             self.example_user("hamlet")])
+        um = exported_usermessages[0]
+        self.assertEqual(len(data["zerver_usermessage"]), len(exported_usermessages))
+        exported_um = find_by_id('zerver_usermessage', um.id)
+        self.assertEqual(exported_um['message'], um.message_id)
+        self.assertEqual(exported_um['user_profile'], um.user_profile_id)
+
+        exported_message = find_by_id('zerver_message', um.message_id)
+        self.assertEqual(exported_message['content'], um.message.content)
+
+        public_stream_names = ['Denmark', 'Rome', 'Scotland', 'Venice', 'Verona']
+        public_stream_ids = Stream.objects.filter(name__in=public_stream_names).values_list("id", flat=True)
+        public_stream_recipients = Recipient.objects.filter(type_id__in=public_stream_ids, type=Recipient.STREAM)
+        public_stream_message_ids = Message.objects.filter(recipient__in=public_stream_recipients).values_list("id", flat=True)
+
+        # Messages from Private Stream C are not exported since no member gave consent
+        private_stream_ids = Stream.objects.filter(name__in=["Private A", "Private B"]).values_list("id", flat=True)
+        private_stream_recipients = Recipient.objects.filter(type_id__in=private_stream_ids, type=Recipient.STREAM)
+        private_stream_message_ids = Message.objects.filter(recipient__in=private_stream_recipients).values_list("id", flat=True)
+
+        pm_recipients = Recipient.objects.filter(type_id__in=consented_user_ids, type=Recipient.PERSONAL)
+        pm_query = Q(recipient__in=pm_recipients) | Q(sender__in=consented_user_ids)
+        exported_pm_ids = Message.objects.filter(pm_query).values_list("id", flat=True).values_list("id", flat=True)
+
+        # Third huddle is not exported since none of the members gave consent
+        huddle_recipients = Recipient.objects.filter(type_id__in=[huddle_a.id, huddle_b.id], type=Recipient.HUDDLE)
+        pm_query = Q(recipient__in=huddle_recipients) | Q(sender__in=consented_user_ids)
+        exported_huddle_ids = Message.objects.filter(pm_query).values_list("id", flat=True).values_list("id", flat=True)
+
+        exported_msg_ids = set(public_stream_message_ids) | set(private_stream_message_ids) \
+            | set(exported_pm_ids) | set(exported_huddle_ids)
+        self.assertEqual(get_set("zerver_message", "id"), exported_msg_ids)
```

Comment:
````
I made these tweaks to the tests just to add an extra assertion:
```
diff --git a/zerver/tests/test_import_export.py b/zerver/tests/test_import_export.py
index 39b1f7b5ff..68ffe4d5b4 100644
--- a/zerver/tests/test_import_export.py
+++ b/zerver/tests/test_import_export.py
@@ -467,7 +467,8 @@ class ImportExportTest(ZulipTestCase):
         create_stream_if_needed(realm, "Private C", invite_only=True)
         self.subscribe(self.example_user("othello"), "Private C")
         self.subscribe(self.example_user("prospero"), "Private C")
-        self.send_stream_message(self.example_email("othello"), "Private C", "Hello Stream C")
+        stream_c_message_id = self.send_stream_message(self.example_email("othello"),
+                                                       "Private C", "Hello Stream C")
 
         # Create huddles
         self.send_huddle_message(self.example_email("iago"), [self.example_email("cordelia"),
@@ -478,9 +479,10 @@ class ImportExportTest(ZulipTestCase):
                                                              self.example_email("othello")])
         huddle_b = Huddle.objects.last()
 
-        self.send_huddle_message(self.example_email("AARON"), [self.example_email("cordelia"),
-                                                               self.example_email("AARON"),
-                                                               self.example_email("othello")])


+        huddle_b_message_id = self.send_huddle_message(
+            self.example_email("AARON"), [self.example_email("cordelia"),
+                                          self.example_email("ZOE"),
+                                          self.example_email("othello")])
 
         # Send message advertising export and make users react
         self.send_stream_message(self.example_email("othello"), "Verona",
@@ -559,6 +561,8 @@ class ImportExportTest(ZulipTestCase):
         exported_msg_ids = set(public_stream_message_ids) | set(private_stream_message_ids) \
             | set(exported_pm_ids) | set(exported_huddle_ids)
         self.assertEqual(get_set("zerver_message", "id"), exported_msg_ids)
+        self.assertNotIn(stream_c_message_id, exported_msg_ids)
+        self.assertNotIn(huddle_b_message_id, exported_msg_ids)
```